docs/advanced-features/event-driven-packaging.md

```diff
@@ -0,0 +1,177 @@
+package in _s3://target-bucket_ (the _target_ bucket may or may not be
+the same as the _source_ bucket).
+3. Each EDP event contains sufficient information for the
+recipient to make Quilt data packages from the event:
+  - S3 Bucket name
+  - Common prefix
+  - Number of files
+  - Timestamp of event
+4. EDP ignores changes to ignored prefixes (`BucketIgnorePrefixes` parameter)
+in the source S3 bucket (e.g. `raw/*`)
+5. EDP publishes events to an AWS EventBridge bus and can forward
+events to external EventBridge buses
+6. Users subscribe the Amazon SNS topic created to recieve notifications
+
+## Example EDP events
+
+Currently, EDP generates two different events:
+
+- `package-objects-ready`
+- `package-ready`
+
+### `package-objects-ready`
+
+This event signals that a batch of files is ready to be packaged.
+For example with `p.set_dir(".",
+f"s3://{e['detail']['bucket']}/{e['detail']['prefix']}")`, followed
+by `p.push()`.
+
+```json
+{
+    "version":"0",
+    "id":"4a26374e-72f2-d78f-071b-dac895697670",
+    "detail-type":"package-objects-ready",
+    "source":"com.quiltdata.edp",
+    "account":"XXXXXXXXXXXX",
+    "time":"2022-12-08T20:01:34Z",
+    "region":"us-east-1",
+    "resources":[
+        "arn:aws:s3:::source-bucket"
+    ],
+    "detail":{
+        "version":"0.1",
+        "bucket":"source-bucket",
+        "prefix":"instrument-name/experiment-id/"
+    }
+}
+```
+
+### `package-ready`
+
+This event signals that a Quilt package is available in an Amazon
+S3 bucket and that a user-specified interval has elapsed to ensure
+that time-driven processes like File Gateway have synchronized their
+state to that of the S3 bucket.
+
+> This is used internally. Most users won't explicitly use this event.
+
+```json
+{
+    "version": "0",
+    "id": "13f2c7fd-1ebd-44c8-406c-fad7204a6b26",
+    "detail-type": "package-ready",
+    "source": "com.quiltdata.edp",
+    "account":"XXXXXXXXXXXX",
+    "time": "2022-12-12T14:28:27Z",
+    "region": "us-east-1",
+    "resources": [
+      "arn:aws:s3:::bucket-name"
+    ],
+    "detail": {
+      "version": "0.1",
+      "package-name": "package/name"
+    }
+}
+```
+
+## Example use case
+
+1. Lab scientist drops files into _s3://RAW/raw/_
+2. Lambda function copies filess to _s3://RAW/other/prefix/_
+3. EDP listens to _s3://RAW/other/*_ and generates a `package-objects-ready`
+event
+4. Second lambda fuction responds to `package-objects-ready` event and
+pushes Quilt data package to _s3://PROCESSED_ bucket
+5. EDP listens for "package created" (from step above), then emits
+a `package_ready` event which is received in a foreign account bus
+6. SNS subscription generates email to scientists
+7. Lab and Computational scientists recieve email notification
+8. Computational scientist opens files backed by _s3://PROCESSED_
+bucket
+
+## Features
+
+- EDP is tolerant of container restarts (e.g. events are not consumed
```

Comment:
This is confusing because it’s an implementation detail about internal containers and has nothing to do with customer containers

Reply:
Recommend eliminating